File: typescript/src/models/delegatePaymentRequest.ts
// this file is @generated
import { type Address, AddressSerializer } from "./address";
import { type Allowance, AllowanceSerializer } from "./allowance";
import { type PaymentMethodCard, PaymentMethodCardSerializer } from "./paymentMethodCard";
import { type RiskSignal, RiskSignalSerializer } from "./riskSignal";

export interface DelegatePaymentRequest {
  paymentMethod: PaymentMethodCard;
  allowance: Allowance;
  billingAddress?: Address | null;
  riskSignals: RiskSignal[];
  metadata: { [key: string]: string };
}

export const DelegatePaymentRequestSerializer = {
  _fromJsonObject(object: any): DelegatePaymentRequest {
    return {
      paymentMethod: PaymentMethodCardSerializer._fromJsonObject(
        object["payment_method"]
      ),
      allowance: AllowanceSerializer._fromJsonObject(object["allowance"]),
      billingAddress: object["billing_address"]
        ? AddressSerializer._fromJsonObject(object["billing_address"])
        : undefined,
      riskSignals: object["risk_signals"].map((item: RiskSignal) =>
        RiskSignalSerializer._fromJsonObject(item)
      ),
      metadata: object["metadata"],
    };
  },

  _toJsonObject(self: DelegatePaymentRequest): any {
    return {
      payment_method: PaymentMethodCardSerializer._toJsonObject(self.paymentMethod),
      allowance: AllowanceSerializer._toJsonObject(self.allowance),
      billing_address: self.billingAddress
        ? AddressSerializer._toJsonObject(self.billingAddress)
        : undefined,
      risk_signals: self.riskSignals.map((item) =>
        RiskSignalSerializer._toJsonObject(item)
      ),
      metadata: self.metadata,
    };
  },
};

File: rust/src/models/delegate_payment_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    address::Address, allowance::Allowance, payment_method_card::PaymentMethodCard,
    risk_signal::RiskSignal,
};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct DelegatePaymentRequest {
    pub payment_method: PaymentMethodCard,
    pub allowance: Allowance,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub billing_address: Option<Address>,
    pub risk_signals: Vec<RiskSignal>,
    pub metadata: std::collections::HashMap<String, String>,
}
